ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh

```diff
@@ -263,6 +263,27 @@ echo "$(date) - waiting for clusteroperators to finish progressing..."
 oc wait clusteroperators --all --for=condition=Progressing=false --timeout=10m
 echo "$(date) - all clusteroperators are done progressing."
 
+# wait up to 10m for the number of nodes to match the number of machines
+MACHINECOUNT=$(kubectl get machines -A --no-headers | wc -l)
+if [ $MACHINECOUNT -gt 0 ]; then
```

Comment:
nit: I don't think you need to special case this, and you can avoid the out-of-loop `get machines` with:

```sh
i=0
while true
do
  MACHINECOUNT=$(kubectl get machines -A --no-headers | wc -l)
  NODECOUNT=$(kubectl get nodes --no-headers | wc -l)
  if [ $MACHINECOUNT -le $NODECOUNT ]
  then
    echo "$(date) - node count ($NODECOUNT) now matches or exceeds machine count ($MACHINECOUNT)"
    break
  fi
  echo "$(date) - $MACHINECOUNT Machines - $NODECOUNT Nodes"
  sleep 30
  ((i++))
  if [ $i -gt 20 ]; then
    echo "Timed out waiting for node count ($NODECOUNT) to equal or exceed machine count ($MACHINECOUNT)."
    exit 1
  fi
fi

In the "no machines" case, that will break out of the loop after the first round.